inference-engine/tests/functional/plugin/gna/Import_export_tests/import_export_memory_layer.cpp

```diff
@@ -0,0 +1,137 @@
+        std::vector<std::string> queryToState;
+        for (const auto &query_state : executableNetwork.QueryState()) {
+            queryToState.push_back(query_state.GetName());
+        }
+        for (const auto &next_memory : importedNetwork.QueryState()) {
+            ASSERT_TRUE(std::find(queryToState.begin(), queryToState.end(), next_memory.GetName()) != queryToState.end())
+                                        << "State " << next_memory.GetName() << " expected to be in memory states but it is not!";
+        }
+    }
+
+protected:
+    void SetUp() override {
+        InferenceEngine::Precision netPrecision;
+        std::tie(netPrecision, targetDevice, exportConfiguration, importConfiguration) = this->GetParam();
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+        auto params = ngraph::builder::makeParams(ngPrc, {{1, 336}});
+        auto mem_c = ngraph::builder::makeConstant(ngPrc, {1, 336}, std::vector<size_t>{1});
+        auto mem_r = std::make_shared<ngraph::opset3::ReadValue>(mem_c, "id");
+
+        auto mul = std::make_shared<ngraph::opset1::Multiply>(params[0], mem_r);
+        auto mem_w = std::make_shared<ngraph::opset3::Assign>(mul, "id");
+
+        auto relu = std::make_shared<ngraph::opset1::Relu>(mul);
+        mem_w->add_control_dependency(mem_r);
+        relu->add_control_dependency(mem_w);
+        ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(relu)};
+        function = std::make_shared<ngraph::Function>(results, params, "ExportImportNetwork");
+    }
+
+private:
+    std::map<std::string, std::string> exportConfiguration;
+    std::map<std::string, std::string> importConfiguration;
+};
+
+TEST_P(ImportMemoryTest, CompareWithRefImpl) {
+    Run();
+};
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::FP16
+};
+
+const std::vector<std::map<std::string, std::string>> exportConfigs = {
+        {
+                {"GNA_DEVICE_MODE", "GNA_SW_EXACT"},
+                {"GNA_SCALE_FACTOR_0", "327.67"}
+        }
+};
+
+const std::vector<std::map<std::string, std::string>> importConfigs = {
+        {
+                {"GNA_DEVICE_MODE", "GNA_SW_EXACT"},
+                {"GNA_SCALE_FACTOR_0", "32767"}
+        },
+        {
+                {"GNA_DEVICE_MODE", "GNA_SW_EXACT"},
+                {"GNA_SCALE_FACTOR_0", "327.67"}
+        },
+};
+
+INSTANTIATE_TEST_CASE_P(smoke_ImportNetworkMemoryCase, ImportMemoryTest,
+                        ::testing::Combine(
+                                ::testing::ValuesIn(netPrecisions),
+                                ::testing::Values(CommonTestUtils::DEVICE_GNA),
+                                ::testing::ValuesIn(exportConfigs),
+                                ::testing::ValuesIn(importConfigs)),
+                        ImportMemoryTest::getTestCaseName);
+
+} // namespace LayerTestsDefinitions
```

Comment:
Please check if it works for GNA_SW_FP32

Reply:
No, It's not working for GNA_SW_FP32